pallets/pool-xyk/src/math.rs

```diff
@@ -192,24 +192,44 @@ impl<T: Config> Pallet<T> {
                 fxw_y_out.clone()
             };
             let nominator = fxw_x * y_out_with_fee.clone();
+            ensure!(fxw_y > y_out_with_fee, Error::<T>::NotEnoughAmountForFee);
```

Comment:
шt will also error in case some of the `FixedWrapper`s already contains errors (not necessarily when there's not enough amount). Seems like it's not expected here

Reply:
I added unwrapping in the check.